src/components/PostInfo.tsx

```diff
@@ -0,0 +1,19 @@
+import React from 'react';
+import { Post } from '../types/Post';
+import { CommentList } from './CommentList';
+import { UserInfo } from './UserInfo';
+import './PostInfo.scss';
+
+type Props = {
+  post: Post;
+};
+
+export const PostInfo: React.FC<Props> = ({ post }) => (
+  <div className="Post">
+    <h3>{post.title}</h3>
+    <p>{post.body}</p>
+    {post.user
+    && <UserInfo user={post.user} />}
+    {post.comments && <CommentList comments={post.comments} />}
```

Comment:
Can post not have comments value? When you create preparedPosts, you set comments value with filter. Filter always returns an array (if don't find coincidences, returns an empty array)